### PR TITLE
Survival analysis time-dependent ROC/AUC

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Survival analysis time-dependent ROC/AUC

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/util.R [hunk 1]
```diff
@@ -2306,3 +2306,9 @@ auroc <- function(score, bool) {
   U  <- sum(rank(score)[!bool]) - n1 * (n1 + 1) / 2
   return(1 - U / n1 / n2)
 }
+
```
Comment (kei51e) [inline R/util.R:2309]: Please add some comments.

File: R/model_eval.R [hunk 2]
```diff
@@ -94,6 +94,18 @@ do_roc_ <- function(df, pred_prob_col, actual_val_col, grid = NULL, with_auc = F
   ret
 }
 
+do_survival_roc_ <- function(df, score_col, time_col, status_col, at = NULL, grid = NULL, with_auc = FALSE, revert = FALSE){
```
Comment (kei51e) [inline R/model_eval.R:97]: Please add comments.

File: R/survival_forest.R [hunk 3]
```diff
@@ -1,3 +1,8 @@
+extract_survival_rate_at <- function(survival, unique_death_times, at) {
```
Comment (kei51e) [inline R/survival_forest.R:1]: Please describe what it is for.

File: R/util.R (hunk 1 above)
Comment (kei51e) [reply R/util.R:2309]: Add 'revert' option to it.

Comment (hideaki) [reply R/util.R:2309]: Done.

File: R/model_eval.R (hunk 2 above)
Comment (hideaki) [reply R/model_eval.R:107]: Done.

File: R/survival_forest.R (hunk 3 above)
Comment (hideaki) [reply R/survival_forest.R:5]: Done.

**Pull request merged**